### ISSUE TITLE
[MAINTENANCE] Book - Pong Tutorial - Bump amethyst to 0.15

### ISSUE THREAD

**Issue by ropewalker** [amethyst/amethyst#2206]
## Description

“[Creating a new project](https://book.amethyst.rs/stable/pong-tutorial/pong-tutorial-01.html)” section of the book refers to version = "0.14" (twice, for Vulcan and for Metal configurations). It should be probably updated to 0.15.

## Reason

Example with 0.14 currently fails with “error[E0433]: failed to resolve: could not find `__rt` in `quote`” in err-derive 0.1.6, while 0.15 uses err-derive 0.2.3 and doesn't seem to have this issue. 

## Impact

I didn't test the full example from the book yet, just the code in “[Drawing the paddles](https://book.amethyst.rs/stable/pong-tutorial/pong-tutorial-02.html)” section, so I am not sure if it is affected.

**Comment by Zooce** [comment]
I'm going through the Pong tutorial now (from the book), and just upgraded to 0.15, which doesn't seem to compile:
```
error: no rules expected the token `#`
   --> /Users/zooce/.cargo/registry/src/github.com-1ecc6299db9ec823/err-derive-0.2.4/src/lib.rs:138:59
    |
138 | decl_derive!([Error, attributes(error, source, cause)] => #[proc_macro_error] error_derive);
    |                                                           ^ no rules expected this token in macro call

error: aborting due to previous error

error: could not compile `err-derive`.
```

**Comment by Zooce** [comment]
FYI: Just finished the Pong tutorial with 0.15.0, and I can confirm that it works.

There were a couple things I noticed -- (I don't think these have anything to do with the Amethyst version)
* There are quite a few unused imports
* While the game is running, if I switch to another application and then switch back, the game is frozen and there are a bunch of these warnings: `[WARN][gfx_backend_metal::window] No frame is available` (on macOS 10.15.4)

**Comment by ChaoticInsanity** [comment]
MacOS: 10.14.6
Rustc: 1.45.0 stable

I just recently started learning Amethyst and I have been working through the 'master' edition of the book. As the book recommends I used the amethyst tools to create the pong application skeleton. I then proceeded to work through section 4.1 and 4.2.

The problem that I have run into is that at the end of section 4.2 my pong application does not render the paddles. There are no errors in the compile or log messages just the same blank screen as section 4.1. The weird thing in this is that if I run the pong_tutorial_2 example the paddles render as expected. I have compared my code to the example code and there is a 100% match.

The only thing that I can come up with is that there is something missing in the Cargo.toml file listed in the book since that is the only difference when compared to the example code.

[INFO][amethyst::app] Initializing Amethyst...
[INFO][amethyst::app] Version: 0.15.0
[INFO][amethyst::app] Platform: x86_64-apple-darwin
[INFO][amethyst::app] Amethyst git commit: UNKNOWN
[INFO][amethyst::app] Rustc version: 1.45.0 Stable
[INFO][amethyst::app] Rustc git commit: 5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2
[WARN][rendy_factory::factory] Slow safety checks are enabled! Disable them in production by enabling the 'no-slow-safety-checks' feature!
[INFO][rendy_util::wrap] Slow safety checks are enabled! You can disable them in production by enabling the 'no-slow-safety-checks' feature!
[INFO][amethyst::app] Engine is shutting down

[pong.zip](https://github.com/amethyst/amethyst/files/4957243/pong.zip)

**Comment by CleanCut** [comment]
>     * While the game is running, if I switch to another application and then switch back, the game is frozen and there are a bunch of these warnings: `[WARN][gfx_backend_metal::window] No frame is available` (at least on macOS 10.15.4)

That behavior is "normal" -- not specific to the pong tutorial.

> FYI: Just finished the Pong tutorial with 0.15.0, and I can confirm that it works (for `metal`).

That's excellent. Can anyone confirm Linux/Windows behavior?

**Comment by CleanCut** [comment]
The actual pong code in the examples directory have been using 0.15 for quite some time, and no complaints from CI or anyone else. I went ahead and did the update myself in #2422
